textdb/textdb-dataflow/src/main/java/edu/uci/ics/textdb/dataflow/keywordmatch/KeywordMatcher.java

```diff
@@ -37,9 +40,10 @@
     private boolean spanSchemaDefined = false;
     private Schema spanSchema;
 
-    public KeywordMatcher(IPredicate predicate, ISourceOperator sourceOperator) {
+    public KeywordMatcher(IPredicate predicate) {
         this.predicate = (KeywordPredicate)predicate;
-        this.sourceOperator = sourceOperator;
+        DataReaderPredicate dataReaderPredicate = this.predicate.convertToDataReaderPredicate();
+        this.sourceOperator = new IndexBasedSourceOperator(dataReaderPredicate);
```

Comment:
Let's leave it as it is.  Currently we haven't specified our own query syntax and we only use a `String` to represent a query. We may need to refactor in the future.